packages/app/src/cli/models/app/app.ts

```diff
@@ -20,13 +20,215 @@ export const LegacyAppSchema = zod
   .strict()
 
 // adding http or https presence and absence of new lines to url validation
-const validateUrl = (zodType: zod.ZodString) => {
+const validateUrl = (zodType: zod.ZodString, {httpsOnly = false, message = 'Invalid url'} = {}) => {
+  const regex = httpsOnly ? /^(https:\/\/)/ : /^(https?:\/\/)/
   return zodType
     .url()
-    .refine((value) => Boolean(value.match(/^(https?:\/\/)/)), {message: 'Invalid url'})
-    .refine((value) => !value.includes('\n'), {message: 'Invalid url'})
+    .refine((value) => Boolean(value.match(regex)), {message})
+    .refine((value) => !value.includes('\n'), {message})
 }
 
+const ensurePathStartsWithSlash = (arg: unknown) => (typeof arg === 'string' && !arg.startsWith('/') ? `/${arg}` : arg)
+const ensureHttpsOnlyUrl = validateUrl(zod.string(), {
+  httpsOnly: true,
+  message: 'Only https urls are allowed',
```

Comment:
Validation would occur on any command where the schema is read in, but usually deploy or push. This exact message could be shown if your `subscription_endpoint_url` is not https, and I've also added it to validate `[webhooks.privacy_compliance]` because thats what we also require on the API and just never added that here.